python/lsst/afw/image/filter.cc

```diff
@@ -40,54 +41,66 @@ using PyFilterProperty = py::class_<FilterProperty, std::shared_ptr<FilterProper
-    clsFilterProperty.def("getLambdaMax", &FilterProperty::getLambdaMax);
-    clsFilterProperty.def_static("reset", &FilterProperty::reset);
-    clsFilterProperty.def_static("lookup", &FilterProperty::lookup, "name"_a);
+            clsFilter.def("__eq__", [](Filter const &self, Filter const &other) { return self == other; },
+                          py::is_operator());
+            clsFilter.def("__ne__", [](Filter const &self, Filter const &other) { return self != other; },
+                          py::is_operator());
+            clsFilter.def_readonly_static("AUTO", &Filter::AUTO);
+            clsFilter.def_readonly_static("UNKNOWN", &Filter::UNKNOWN);
+            clsFilter.def("getId", &Filter::getId);
+            clsFilter.def("getName", &Filter::getName);
+            clsFilter.def("getCanonicalName", &Filter::getCanonicalName);
+            clsFilter.def("getAliases", &Filter::getAliases);
+            clsFilter.def("getFilterProperty", &Filter::getFilterProperty);
+            clsFilter.def_static("reset", &Filter::reset);
+            clsFilter.def_static("define", &Filter::define, "filterProperty"_a, "id"_a = Filter::AUTO,
+                                 "force"_a = false);
+            clsFilter.def_static("defineAlias", &Filter::defineAlias, "oldName"_a, "newName"_a, "force"_a = false);
+            clsFilter.def_static("getNames", &Filter::getNames);
+        });
+}
 
-    PyFilter clsFilter(mod, "Filter");
-    clsFilter.def(py::init<std::string const &, bool const>(), "name"_a, "force"_a = false);
-    clsFilter.def(py::init<int>(), "id"_a = Filter::UNKNOWN);
-    clsFilter.def(py::init<std::shared_ptr<daf::base::PropertySet const>, bool const>(), "metadata"_a,
-                  "force"_a = false);
-    clsFilter.def("__eq__", [](Filter const &self, Filter const &other) { return self == other; },
-                  py::is_operator());
-    clsFilter.def("__ne__", [](Filter const &self, Filter const &other) { return self != other; },
-                  py::is_operator());
-    clsFilter.def_readonly_static("AUTO", &Filter::AUTO);
-    clsFilter.def_readonly_static("UNKNOWN", &Filter::UNKNOWN);
-    clsFilter.def("getId", &Filter::getId);
-    clsFilter.def("getName", &Filter::getName);
-    clsFilter.def("getCanonicalName", &Filter::getCanonicalName);
-    clsFilter.def("getAliases", &Filter::getAliases);
-    clsFilter.def("getFilterProperty", &Filter::getFilterProperty);
-    clsFilter.def_static("reset", &Filter::reset);
-    clsFilter.def_static("define", &Filter::define, "filterProperty"_a, "id"_a = Filter::AUTO,
-                         "force"_a = false);
-    clsFilter.def_static("defineAlias", &Filter::defineAlias, "oldName"_a, "newName"_a, "force"_a = false);
-    clsFilter.def_static("getNames", &Filter::getNames);
+PYBIND11_MODULE(filter, mod) {
+    lsst::utils::python::WrapperCollection wrappers(mod, "lsst.afw.image.filter");
+    wrappers.addInheritanceDependency("lsst.daf.base");
```

Comment:
Should just be signature dependency.

Reply:
For now this just forces the py_::import. Fixing this in the afw.image ticket.